packages/xod-client/src/core/containers/App.jsx

```diff
@@ -186,14 +186,15 @@ export default class App extends React.Component {
   }
 
   renderPopupProjectPreferences() {
-    return (
+    return this.props.popups.projectPreferences ? (
       <PopupProjectPreferences
-        isVisible={this.props.popups.projectPreferences}
+        isVisible
```

Comment:
Previously, the popup component was mounted right on the running of the application. So it needs a subscription on changes in the project. And it was okay since we're changing everything only within the state of the popup and apply changes by clicking "Update properties" button.
But now we have to update `apiKey` property of the project, so it updates the project and it will update the state of the popup and all unsaved changes might be lost. To avoid this, I just mount the component only when it needed ("Show Project Properties" clicked). At this moment, it receives the actual project, so we don't need to subscribe to project changes (except a change of `apiKey`) and there is no conflict between state and props.